tree-sitter-dscexpression/grammar.js

```diff
@@ -29,7 +29,7 @@ module.exports = grammar({
     _quotedString: $ => seq('\'', $.string, '\''),
     // ARM strings do not allow to contain single-quote characters
     string: $ => /[^']*/,
-    number: $ => /-?\d+/,
+    number: $ => /-?(\d*[.])?\d+/,
```

Comment:
If we want to prevent `.1`, then we just need to require at least one digit, so something like:

```
^-?\d+([.]\d+)?$
```

See https://regex101.com/r/lynDYN/1 as my tests

Reply:
If we add the check to the Rust parser, then the grammar can stay as it was before!
I kept the tree-sitter tests to make sure the floating point numbers result in error nodes.